app/models/miq_request_task/post_install_callback.rb

```diff
@@ -3,12 +3,9 @@ module MiqRequestTask::PostInstallCallback
 
   module ClassMethods
     def post_install_callback(id)
-      p = find_by_id(id.to_i)
-      if p
-        p.provision_completed_queue
-      else
-        _log.warn("task_id=#{id.inspect} not found")
-      end
+      find(id).provision_completed_queue
+    rescue ActiveRecord::RecordNotFound
+      _log.warn("task_id=#{id.inspect} not found")
```

Comment:
"Expecting exceptions" is needlessly more expensive. Simply changing the `find_by_id` -> `find_by(:id =>` and leaving the `nil` is more appropriate, I think.